modules/evm/rpc/src/lib.rs

```diff
@@ -190,12 +190,43 @@ where
 		}
 	}
 
-	fn estimate_resources(
-		&self,
-		extrinsic: Bytes,
-		request: CallRequest,
-		_: Option<B>,
-	) -> Result<EstimateResourcesResponse> {
+	fn estimate_resources(&self, extrinsic: Bytes, _: Option<B>) -> Result<EstimateResourcesResponse> {
+		let utx = mandala_runtime::UncheckedExtrinsic::decode(&mut &*extrinsic).map_err(|_| Error {
```

Comment:
this needs to be generic otherwise we can't use it on other runtime

One way is to have a EvmCallMatcher trait and each runtime will provide its implementation